src/metldata/builtin_transformations/infer_relations/data_transform.py

```diff
@@ -148,7 +145,7 @@ def resolve_path_element(
     data: DataPack,
     source_resource_id: ResourceId,
     path_element: RelationPathElement,
-) -> set[ResourceId]:
+) -> set[ResourceId] | frozenset[ResourceId]:
```

Comment:
This can be simplified to just `frozenset[ResourceId]` if the changes for both functions above are implemented